app/core/blockchain/state_updater.py

```diff
@@ -468,7 +461,87 @@ def pay_fee_for_transaction(cur, transaction, creator):
         )
     return True
 
+def handle_sc_transaction(cur, transaction, creator_wallet, newblockindex):
+    
+    transaction_all = transaction
+    transaction = transaction['transaction']
+    transaction_data = transaction['specific_data']
+    while isinstance(transaction_data, str):
+            transaction_data = json.loads(transaction_data)
+            transaction['specific_data']=transaction_data
+
+    transaction_code = transaction['transaction_code'] if 'transaction_code' in transaction else transaction[
+            'trans_code']
+    
+    sc_txns_simplified = simplify_transactions_sc(cur = cur, transaction= transaction_all)
+    sc_nesting = 0  # Denotes the level of nesting of smart contract call. 1 for a normal sc call. 2 for sc calling sc. 
+    sc_in_failed_state = False  # Used to flag a sc execution failure for flushing out all future child transactions
+    for transaction in sc_txns_simplified:
+
+        if sc_in_failed_state:  # If any transaction in SC fails, keep flushing future transactions till all parent SCs end
+            if transaction == 'SC_END':
+                sc_nesting -= 1
+                if sc_nesting == 0:
```

Comment:
if we're processing sc child txn only in this function, then won't need sc_nesting = 0 right?

Reply:
we are not processing child txn alone but tree of childs of an sc seperatly. We need roll back if one in that batch doesnt work.